src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -584,4 +603,15 @@ internal interface IOperationStatus
 
         public bool IsSuccess { get; }
     }
+
+    /// <summary>
+    /// Used specifically for UDF, so that we can set the texlnode for UDF body that was already parsed.
+    /// </summary>
+    internal class UDFCheckResult : CheckResult

```

Comment:
>UDFCheckResult [](http://example.com/codeflow?start=19&length=14)

Why a new derived class? It's not adding anything.

Reply:
Setting TexlNode is very specific to UDFs and we don't want to allow setting that from anywhere else other then from UDFCheckResult.

Reply:
Why can't that be in the constructor of CheckResult?

Reply:
wrap in a ParseResult and call the existing SetText() method. 

Else every caller of CheckResult has to now special case TexlNode vs. ParseResult.